### PR TITLE
Expose the drop-disguise-when-attacking aspect to yaml.

### PR DESCRIPTION
**Pull request by GraionDilach**
Also adds Damaged for more usecases.

### CODE REVIEW

File: OpenRA.Mods.Cnc/Traits/Disguise.cs [hunk 1]
```diff
@@ -68,10 +76,13 @@ class DisguiseInfo : ITraitInfo
 		[Desc("The condition to grant to self while disguised.")]
 		public readonly string DisguisedCondition = null;
 
+		[Desc("Triggers which cause the actor to drop it's disguise. Possible values: None, Attack, Damaged.")]
+		public RevealDisguiseType RevealDisguiseTriggers = RevealDisguiseType.Attack;
```
Comment (pchote) [inline OpenRA.Mods.Cnc/Traits/Disguise.cs]: Can we please rename this to `RevealDisguiseOn` for consistency with Cloak?

Comment (GraionDilach) [inline OpenRA.Mods.Cnc/Traits/Disguise.cs]: Done.

**Review by reaperrr — APPROVED** [review]

File: OpenRA.Mods.Cnc/Traits/Disguise.cs [hunk 2]
```diff
@@ -68,10 +76,13 @@ class DisguiseInfo : ITraitInfo
 		[Desc("The condition to grant to self while disguised.")]
 		public readonly string DisguisedCondition = null;
 
+		[Desc("Triggers which cause the actor to drop it's disguise. Possible values: None, Attack, Damaged.")]
+		public RevealDisguiseType RevealDisguiseOn = RevealDisguiseType.Attack;
```
Comment (reaperrr) [inline OpenRA.Mods.Cnc/Traits/Disguise.cs]: This can/should be readonly.

Comment (GraionDilach) [inline OpenRA.Mods.Cnc/Traits/Disguise.cs]: Fixed, thanks.

**Review by pchote — APPROVED** [review]

**Comment by pchote** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/ad30509f8a5195a627be9079c15efa6aff9f3281

**Pull request merged**